api/envoy/extensions/filters/http/proto_api_scrubber/v3/config.proto

```diff
@@ -0,0 +1,323 @@
+  enum FilteringMode {
+    // If unspecified, falls back to ``OVERRIDE``.
+    FILTERING_MODE_UNSPECIFIED = 0;
+
+    // Override the original request/response body with the filtered
+    // request/response body.
+    OVERRIDE = 1;
+  }
+
+  // The proto descriptor set for the proto services.
+  oneof descriptor_set {
+    // It could be passed by a local file through ``Datasource.filename`` or
+    // embedded in the ``Datasource.inline_bytes``.
+    config.core.v3.DataSource data_source = 1;
+
+    // Unimplemented, the key of proto descriptor TypedMetadata.
+    // Among filters depending on the proto descriptor, we can have a
+    // TypedMetadata for proto descriptors, so that these filters can share one
+    // copy of proto descriptor in memory.
+    string proto_descriptor_typed_metadata = 2;
+  }
+
+  // Data sources for the request entitlements.
+  // Entitlements are matched with restrictions of the same name at runtime.
+  // Make sure that the entitlement source is present for each of the
+  // corresponding restrictions.
+  // Key - entitlement name
+  // Value - entitlement source
+  map<string, RequestEntitlementSource> request_entitlement_sources = 3;
+
+  // Contains the restrictions for the API, methods, messages, fields, etc.
+  Restrictions restrictions = 4;
+
+  FilteringMode filtering_mode = 5;
+}
+
+// Specifies the source of entitlements for a request.
+// The value for each entitlement is expected to be a comma separated list of
+// strings.
+message RequestEntitlementSource {
+  enum Source {
+    // If unspecified, the entitlement will be empty string.
+    SOURCE_UNSPECIFIED = 0;
+
+    // The entitlement will be fetched from the request header ``key``.
+    REQUEST_HEADER = 1;
+
+    // The entitlement will be fetched from the filter state ``key``.
+    FILTER_STATE = 2;
+  }
+
+  Source source = 1;
+
+  // The key to lookup the entitlement from the specified ``source``.
+  string key = 2;
+}
+
+// Contains the restrictions for the API, methods, messages, fields, etc.
+message Restrictions {
+  // Specifies the method restrictions.
+  // Key - Fully qualified method name.
+  //     - ``${package}.${Service}.${Method}``, like
+  //     - ``endpoints.examples.bookstore.BookStore.GetShelf``
+  // Value - Method restrictions.
+  map<string, MethodRestrictions> method_restrictions = 1;
+}
+
+// Contains the method restrictions which include the method level restrictions
+// as well as the field level restrictions for the request and response fields.
+message MethodRestrictions {
+  // Maps restriction types to its values.
+  message RestrictionMap {
+    // Key - Restriction type.
+    // Value - Restriction values.
+    map<string, Restriction> restrictions = 1;
+  }
+
+  // Restrictions that apply to request fields of the method.
+  // Key - field mask like path of the field eg, foo.bar.baz
+  // Value - Restrictions map containing the mapping from restriction type to
+  // the restriction values.
```

Comment:
```suggestion
  // Restrictions that apply to request fields of the method:
  // - Key - field mask like path of the field eg, ``foo.bar.baz``
  // - Value - Restrictions map containing the mapping from restriction type to
  // the restriction values.
```

with these - not sure if we want a list but currently this would all just render on one line